src/Uno.UI/Microsoft/UI/Xaml/Controls/RatingControl/RatingControl.cs

```diff
@@ -0,0 +1,1195 @@
+using RatingControlAutomationPeer = Microsoft.UI.Xaml.Automation.Peers.RatingControlAutomationPeer;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	public partial class RatingControl : Control
+	{
+		private const float c_horizontalScaleAnimationCenterPoint = 0.5f;
+		private const float c_verticalScaleAnimationCenterPoint = 0.8f;
+		private readonly Thickness c_focusVisualMargin = new Thickness(-8, -7, -8, 0);
+		private const int c_defaultRatingFontSizeForRendering = 32; // (32 = 2 * [default fontsize] -- because of double size rendering), remove when MSFT #10030063 is done
+		private const int c_defaultItemSpacing = 8;
+
+		private const float c_mouseOverScale = 0.8f;
+		private const float c_touchOverScale = 1.0f;
+		private const float c_noPointerOverMagicNumber = -100;
+
+		// 22 = 20(compensate for the -20 margin on StackPanel) + 2(magic number makes the text and star center-aligned)
+		private const float c_defaultCaptionTopMargin = 22;
+
+		private const int c_noValueSetSentinel = -1;
+
+		private static UISettings _uiSettings = null;
+
+		private DispatcherHelper m_dispatcherHelper;
+
+		// Private members
+		TextBlock m_captionTextBlock;
+
+		CompositionPropertySet m_sharedPointerPropertySet;
+
+		private StackPanel m_backgroundStackPanel;
+		private StackPanel m_foregroundStackPanel;
+
+		private bool m_isPointerOver = false;
+		private bool m_isPointerDown = false;
+		private double m_mousePercentage = 0.0;
+
+		private RatingInfoType m_infoType = RatingInfoType.Font;
+		private double m_preEngagementValue = 0.0;
+		private bool m_disengagedWithA = false;
+		private bool m_shouldDiscardValue = true;
+		private long m_fontFamilyChangedToken;
+
+		public RatingControl()
+		{
+			m_dispatcherHelper = new DispatcherHelper(this);
+			//__RP_Marker_ClassById(RuntimeProfiler.ProfId_RatingControl);
+
+			DefaultStyleKey = typeof(RatingControl);
+		}
+
+		~RatingControl()
+		{
+			// We only need to use safe_get in the deruction loop
+			RecycleEvents(true /* useSafeGet */);
+		}
+
+		public event TypedEventHandler<RatingControl, object> ValueChanged;
+
+		private float RenderingRatingFontSize()
+		{
+			// MSFT #10030063 Replacing with Rating size DPs
+			return (float)(c_defaultRatingFontSizeForRendering * GetUISettings().TextScaleFactor);
+		}
+
+		private float ActualRatingFontSize()
+		{
+			return RenderingRatingFontSize() / 2;
+		}
+
+		// TODO MSFT #10030063: Convert to itemspacing DP
+		private double ItemSpacing()
+		{
+			// Stars are rendered 2x size and we use expression animation to shrink them down to desired size,
+			// which will create those spacings (not system margin).
+			// Since text scale factor won't affect system margins,
+			// when stars get bigger, the spacing will become smaller.
+			// Therefore we should include TextScaleFactor when calculating item spacing
+			// in order to get correct total width and star center positions.
+			double defaultFontSize = c_defaultRatingFontSizeForRendering / 2;
```

Comment:
Codacy found an issue: [Cast one of the operands of this division to 'double'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6162540)